src/appleseed.studio/mainwindow/rendering/materialdrophandler.cpp

```diff
@@ -43,20 +45,22 @@ namespace appleseed {
 namespace studio {
 
 MaterialDropHandler::MaterialDropHandler(
-    const renderer::Project&    project,
-    RenderingManager&           rendering_manager)
-  : m_project(project), m_rendering_manager(rendering_manager)
+    const renderer::Project&        project,
+    RenderingManager&               rendering_manager,
+    const renderer::ScenePicker&    scene_picker)
+  : m_project(project), m_rendering_manager(rendering_manager), m_scene_picker(scene_picker)
 {
 }
 
 void MaterialDropHandler::slot_material_dropped(
-    const foundation::Vector2d& drop_pos,
-    const QString&          material_name)
+    const foundation::Vector2d&     drop_pos,
+    const QString&                  material_name)
 {
     m_drop_pos = drop_pos;
     m_material_name = material_name.toStdString();
 
-    if (!m_project.has_trace_context())
+    // Tell the user to render once, if assembly tree hasn't been built yet.
+    if (m_project.get_trace_context().get_assembly_tree().get_nodes().size() == 0)
```

Comment:
Why this change? What's the problem with `has_trace_context()`?

Reply:
`m_project.has_trace_context()` will always return true, since it gets created with the `ScenePicker` which in turn gets created with the `RenderTab`